controller/contour/controller.go

```diff
@@ -112,6 +113,9 @@ func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha
 		return fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
 			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
 	}
+	if err := r.ensureJob(ctx, contour); err != nil {
```

Comment:
The real dependency here is whether the TLS secrets are present, so perhaps we should check for those first, then use certgen as a secret generating strategy. I don't mind leaving this as is for now, but add some comments.

Reply:
added code docs.